<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1140</width>
    <height>474</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3" stretch="3,1">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="8,0">
      <item>
       <widget class="QWidget" name="widget" native="true">
        <widget class="matrixWidget" name="LEDmatrixWidget" native="true">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>960</width>
           <height>240</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="menuButtons" native="true">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QPushButton" name="clearDisplay">
           <property name="text">
            <string>Clear Display</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="invertDisplay">
           <property name="text">
            <string>Invert Display</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="liveView">
           <property name="text">
            <string>Live View</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="sendtoScreen">
           <property name="text">
            <string>Send to Screen</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>30</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="savePreset">
           <property name="text">
            <string>Save Preset</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="openPreset">
           <property name="text">
            <string>Open Preset</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QWidget" name="animToolWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="8,1">
         <item>
          <widget class="QSlider" name="horizontalSlider">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>0</number>
           </property>
           <property name="sliderPosition">
            <number>0</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="frameLabel">
           <property name="text">
            <string>frame/frame</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="8,8,8,8,8,5">
         <item>
          <widget class="QRadioButton" name="toggleLoop">
           <property name="text">
            <string>Loop</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="prevFrame">
           <property name="text">
            <string>&lt;&lt; Previous frame</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="playpauseButton">
           <property name="text">
            <string>Play</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="nextFrame">
           <property name="toolTip">
            <string>Creates a new frame if necessary</string>
           </property>
           <property name="toolTipDuration">
            <number>1</number>
           </property>
           <property name="text">
            <string>Next frame &gt;&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="frame_2">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="deleteFrame">
           <property name="text">
            <string>Delete frame</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1140</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <addaction name="actionConnect"/>
    <addaction name="actionDisconnect"/>
    <addaction name="separator"/>
    <addaction name="actionMicrphone_monitor"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionAnimation"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuOptions"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionAnimation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Animation</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>Connect</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
  <action name="actionSet_Frame_from_file">
   <property name="text">
    <string>Set Frame from file</string>
   </property>
  </action>
  <action name="actionMicrphone_monitor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Micrphone monitor</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>matrixWidget</class>
   <extends>QWidget</extends>
   <header location="global">matrixwidget.h</header>
   <container>1</container>
   <slots>
    <slot>slotClear()</slot>
    <slot>slotInvert()</slot>
    <slot>slotSave()</slot>
    <slot>slotOpen()</slot>
    <slot>slotSendOut()</slot>
    <slot>slotSetIP(QString)</slot>
    <slot>slotSliderValChanged(int)</slot>
    <slot>slotDeleteFrame()</slot>
    <slot>slotCopyFrame()</slot>
    <slot>slotPasteFrame()</slot>
    <slot>slotMic(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionConnect</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>slotConnect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>551</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotSliderValChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>502</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteFrame</sender>
   <signal>clicked()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotDeleteFrame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1061</x>
     <y>374</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCopy</sender>
   <signal>triggered()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotCopyFrame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPaste</sender>
   <signal>triggered()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotPasteFrame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionMicrphone_monitor</sender>
   <signal>toggled(bool)</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotMic(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearDisplay</sender>
   <signal>clicked()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotClear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1078</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>invertDisplay</sender>
   <signal>clicked()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotInvert()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1078</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sendtoScreen</sender>
   <signal>clicked()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotSendOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1078</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>savePreset</sender>
   <signal>clicked()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1078</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openPreset</sender>
   <signal>clicked()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1078</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>LEDmatrixWidget</receiver>
   <slot>slotOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>IPchanged(QString)</signal>
  <slot>slotConnect()</slot>
  <slot>slotMic(bool)</slot>
 </slots>
</ui>
